components/Navbar.js

```diff
@@ -0,0 +1,134 @@
+import styled from 'styled-components';
+import Link from 'next/link';
+import { useRouter } from 'next/router';
+import MenuIcon from '../public/assets/hamburger-menu.svg';
+import CloseIcon from '../public/assets/close-icon.svg';
+import { useState } from 'react';
+
+export default function Navbar({ storages }) {
+  const router = useRouter();
+  const [isMenuActive, setIsMenuActive] = useState(false);
+  const isactive = {
+    yes: '#Ff3c1a',
+    no: 'transparent',
+  };
+
+  return (
+    <StyledNav>
+      <StyledButton
+        onClick={() => {
+          setIsMenuActive((curr) => !curr);
```

Comment:
could you explain this to me? What is curr, because I don't see it anywhere else in your code... I think I expected setIsMenuActive(!false) or so